src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -847,7 +847,7 @@ public JoinData getJoinData() {
 
     @Override
     public Value<Instant> lastPlayed() {
-        return new SpongeValue<>(Keys.LAST_DATE_PLAYED, Instant.EPOCH, Instant.now());
+        return new SpongeValue<>(Keys.LAST_DATE_PLAYED, Instant.EPOCH, SpongePlayerDataHandler.getLastPlayed(this.getUniqueID()).get());
```

Comment:
I don't think this is the way to fix it. `lastPlayed` should indicate when the player indeed last played, which in the case of an online player is _right now_. This also breaks the expectation of other plugins that might use this, so even though this isn't an API change, it's a behavioral change and we should not be doing that either with good reason.

For a simple patch, setting some flag on the player (or other) object that states a player is on its first connection would be better, which would then get passed up through the API.

From what @ryantheleach said on the linked issue (to which I agree)

> LastJoined should be when a player last connected (successfully)
> LastPlayed should be the current instant, if the player is currently online.

Thus, the better solution here would to be to add a `Keys.LAST_DATE_JOINED` key which gets populated with the last join date. That should then be used for comparisons.